### DISCUSSION TITLE
Custom npm registry for custom npm packages?

### DISCUSSION THREAD

**Discussion by HenrikPoulsen** [renovatebot/renovate#12312] [Request Help] [answered]
I have the following scenario:
At the company we have many in house developed npm packages that live in multiple different custom npm registries we host in artifactory.
Many of these packages are command line packages that we install across the company in hundreds of repos in various CI solutions using a variant of `npm install <packagename>@<packageversion> --registry https://customregistry` which lives in CI yaml files or shell scripts etc inside the repos.
I would like to be able to renovate these instances with one or more custom regex managers to cover these invocations.
The problem is however that the npm datasource does not allow for custom registries and thus will go to npmjs to look for these packages that do not exist here.
Is there some way we can get this to be a supported use case?

**Comment by rarkins** [comment, marked as answer]
What about a packageRule with `npmrc` field including a `registry=....` line?

**Reply by HenrikPoulsen** [reply]
Seems to work. I'll go with that. Thanks
